### PR TITLE
Mainnet pool reward contributions.

### PR DESCRIPTION
**Pull request by ice-myles**
Mainnet pool reward contributions.

### CODE REVIEW

**Comment by ice-cronus** [comment]
I think we need those new fields in history as well, cuz we cannot rely on 0.3 of ethereum... fields, cuz value is changeable

File: coin-distribution/pending_review.go [hunk 1]
```diff
@@ -278,6 +278,31 @@ func (r *repository) CollectCoinDistributionsForReview(ctx context.Context, reco
...
+		int64(record.Balance*100),
+		record.Username,
+		record.ReferredByUsername,
+		record.UserID,
+		record.EarnerUserID,
+		record.EthAddress)
+	sql := `INSERT INTO coin_distributions_by_earner(created_at,day,internal_id,balance,username,referred_by_username,user_id,earner_user_id,eth_address) 
+											 VALUES ($1,        $2, $3,         $4,     $5,      $6,                  $7,     $8,            $9)
+						ON CONFLICT (day, user_id, earner_user_id) DO UPDATE
+							SET balance = coin_distributions_by_earner.balance + EXCLUDED.balance`
```
Comment (ice-cronus) [inline coin-distribution/pending_review.go]: I am aware if we can get duplicated increment here, but we need it to sum values across batches anyway... But at least we can to do it with the same insertion batch in CollectCoinDistributionsForReview to keep it atomic with other values or to move that sum logic to code level somehow

File: coin-distribution/pending_review.go [hunk 2]
```diff
@@ -278,6 +278,31 @@ func (r *repository) CollectCoinDistributionsForReview(ctx context.Context, reco
 	return errors.Wrapf(err, "failed to insert into coin_distributions_by_earner [%v]", len(records))
 }
 
+func (r *repository) CollectMainnetPoolRewardContributionsForReview(ctx context.Context, record *ByEarnerForReview) error {
```
Comment (ice-ares) [inline coin-distribution/pending_review.go]: No, reuse the same api

File: coin-distribution/contract.go [hunk 3]
```diff
@@ -37,6 +37,7 @@ type (
 		NotifyCoinDistributionCollectionCycleEnded(ctx context.Context) error
 		GetCollectorSettings(ctx context.Context) (*CollectorSettings, error)
 		CollectCoinDistributionsForReview(ctx context.Context, records []*ByEarnerForReview) error
+		CollectMainnetPoolRewardContributionsForReview(ctx context.Context, records *ByEarnerForReview) error
```
Comment (ice-ares) [inline coin-distribution/contract.go]: Reuse same api

File: cmd/freezer/api/swagger.yaml [hunk 4]
```diff
@@ -78,6 +78,9 @@ definitions:
       total:
         example: 1,243.02
         type: string
+      totalMainnnetRewardPoolContributions:
```
Comment (ice-ares) [inline cmd/freezer/api/swagger.yaml]: Mainnet

And remove the "s" at the end

**Review by ice-ares — CHANGES_REQUESTED** [review]

**Pull request closed without merging**